src/storage/src/sst.rs

```diff
@@ -80,13 +86,22 @@ impl LevelMetas {
     pub fn merge(&self, files_to_add: impl Iterator<Item = FileHandle>) -> LevelMetas {
         let mut merged = self.clone();
         for file in files_to_add {
-            let level = file.level_index();
-
-            merged.levels[level].add_file(file);
+            let level = file.level();
+            merged.levels[level as usize].add_file(file);
         }
+        merged
+    }
 
-        // TODO(yingwen): Support file removal.
-
+    /// Removes files with given file meta and builds a new [LevelMetas].
+    ///
+    /// # Panics
+    /// Panics if level of [FileHandle] is greater than [MAX_LEVEL].
+    pub fn remove(&self, files_to_remove: impl Iterator<Item = FileMeta>) -> LevelMetas {
```

Comment:
We should add the `files_to_remove` param to `merge()` to make the file addition and removal atomic.